.github/docker_images/cmake_build_versions/Dockerfile

```diff
@@ -0,0 +1,33 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+FROM amazonlinux:2023
+
+ARG CMAKE_VERSION
+ARG CMAKE_DOWNLOAD_URL
+ARG CMAKE_SHA256
+
+VOLUME ["/awslc"]
+
+RUN dnf install -y tar ninja-build golang gcc gcc-c++ libcurl-devel libarchive-devel zlib-devel xz-devel
```

Comment:
I'm not familiar with `dnf`. Is there some analog to `apt update` or `yum update` that we should run before installing?

Reply:
`dnf` is technically the replacement for `yum` that occurred in Amazon Linux 2023. Not that you can't still use the `yum` commands. https://docs.aws.amazon.com/linux/al2023/ug/compare-with-al2.html